### PR TITLE
feat(linter): allow lint rules with the same name

### PR DESCRIPTION
**Pull request by camc314**
Apologies @Boshen for the stupidly large diff.

I've done the following:

 - `RuleEnum`'s members are now prefixed with the plugin name. e.g. `NoDebugger` has become `EslintNoDebugger`
- updated tester.rs to accept the rule's NAME, CATEGORY to allow us to test rules with the same name (diff category)
- updates `declare_all_lint_rules` satisfy the first above change.

### CODE REVIEW

**Comment by camc314** [comment]
* **#7496** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7496?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7496?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Comment by Boshen** [comment]
### Merge activity

* **Nov 26, 11:24 PM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

**Comment by Boshen** [comment]
cc @Sysix this may break oxlint plugin.

File: crates/oxc_linter/src/rules/import/no_namespace.rs [hunk 1]
```diff
@@ -7,22 +7,22 @@ use oxc_syntax::module_record::ImportImportName;
 
 use crate::{context::LintContext, rule::Rule};
 
-fn import_no_namespace_diagnostic(span: Span) -> OxcDiagnostic {
+fn no_namespace_diagnostic(span: Span) -> OxcDiagnostic {
```
Comment (pumano) [inline crates/oxc_linter/src/rules/import/no_namespace.rs:10]: Looks like entire filename can be renamed to "no-namespace.rs" too

**Pull request merged**